Feature/table validation example

ektasawant opened the pull request:
<!-- If this pull request fixes an Issue, link it below. If not, you can remove the line below -->
Fixes PXBLUE-2322 .

<!-- Include a bulleted list summarizing the main changes you have made in this PR -->
#### Changes proposed in this Pull Request:
- Form validation Table view (Desktop)
- Form Validation List View (Mobile)

<!-- Instruction for PR reviewers, if more complicated than a simple yarn start -->
#### To Test:
- Local Test URL: http://localhost:4200/forms-and-validation/in-a-table

ektasawant commented on src/app/pages/forms-and-validation/in-a-list/in-a-list.module.ts:
```diff
@@ -22,6 +23,7 @@ import { InAListComponent } from './in-a-list.component';
         BrowserAnimationsModule,
```
after removing `BrowserAnimationsModule` getting following error:

![image](https://user-images.githubusercontent.com/10433274/131313906-fabd3e3d-2431-4419-ae9c-bf778edfca71.png)

emclaug2 replied:
Ah okay my mistake, I didn't run the tests.

ektasawant commented:
> I do think the switch between table and list is happening prematurely. I'd match react here and switch at the `xs` (600px) viewport.

Done Please review.

emclaug2 commented:
Thanks for updating the breakpoint changes, this looks really good.  There's just one more small thing that I noticed whenever I refactored the react example this morning.  The margin above the divider of the last "Max" InfoListItem should be increased a little bit.
![image](https://user-images.githubusercontent.com/6538289/131561479-f3b75e6d-6887-455d-8b77-55118119cd40.png)


I handled this on the react side by introducing a 4th InfoListItem that has an empty `title` prop so it renders empty, but has the `partial` divider.  Then you can easily set a marginTop/marginBottom to get the alignment to match the specs.

ektasawant commented:
> Thanks for updating the breakpoint changes, this looks really good. There's just one more small thing that I noticed whenever I refactored the react example this morning. The margin above the divider of the last "Max" InfoListItem should be increased a little bit.
> ![image](https://user-images.githubusercontent.com/6538289/131561479-f3b75e6d-6887-455d-8b77-55118119cd40.png)
> 
> I handled this on the react side by introducing a 4th InfoListItem that has an empty `title` prop so it renders empty, but has the `partial` divider. Then you can easily set a marginTop/marginBottom to get the alignment to match the specs.

I have fixed following, please review

1. Alignment of App bar title with InfoListItem title
2. Add empty InfoListItem for matching specs.

emclaug2 commented:
Looking good!
